src/main/kotlin/se/uulm/snowballr/backend/repository/VerificationTokenTableRepo.kt

```
@@ -49,11 +48,7 @@ class VerificationTokenTableRepo(
     }
 
     override suspend fun getVerificationTokenByValue(token: String): VerificationToken? = db.query {
-        VerificationTokenTable
-            .selectAll()
-            .where { VerificationTokenTable.token eq token }
-            .map { it.toVerificationToken() }
-            .singleOrNull()
+        VerificationTokenTable.getEntityOrNull(ResultRow::toVerificationToken) { VerificationTokenTable.token eq token }
```

I think this method could return a Result as well instead of returning null in the error case which leads to throwing an exception on the service layer. What do you think?

---

Yes, and no.

If I understood it correctly, the result thing would only return a `NotFoundException`.
If used the new method with `getOrThrow` and still want to return a `VerificationTokenNotFound` exception, e would need to wrap this in a `try`-`catch` block, which would only bloat the code.

Since this function is only used here, I think this is okay.

---

Actually no, then you misunderstood the ResultType. You can wrap every exception you want.

```kotlin
override suspend fun getVerificationTokenByValue(token: String): Result<VerificationToken> = db.query {
        val token = VerificationTokenTable.getEntityOrNull(ResultRow::toVerificationToken) { VerificationTokenTable.token eq token }

        return@query if (token != null) {
            Result.success(token)
        } else {
            Result.failure(SnowballRException.VerificationTokenNotFoundException())
        }
        
    }
```

This is an example how this function then could look like. Do you think this refactoring would improve the code or not? Or should be keep it for now and discuss such a scenario in another issue, as we will probably only use this function once?!

---

No, I didn't misunderstood the Result Type. But I was so focused on refactoring the method using the new helper methods (which throw NotFoundException), that I didn't think about simply doing it directly.

Either way this doesn't really improve the code just moves the focus into a different section.
You can write this down and we will discuss this later, but for now I think we can keep it like it is.

---

All right, that's how it's done.